src/rest/resource.cpp

```diff
@@ -312,17 +312,66 @@ void Resource::GetDataState(Response &aResponse) const
     aResponse.SetResponsCode(errorCode);
 }
 
-void Resource::State(const Request &aRequest, Response &aResponse) const
+void Resource::SetDataState(const Request &aRequest, Response &aResponse) const
 {
+    otbrError   error = OTBR_ERROR_NONE;
     std::string errorCode;
+    std::string body;
 
-    if (aRequest.GetMethod() == HttpMethod::kGet)
+    VerifyOrExit(Json::JsonString2String(aRequest.GetBody(), body), error = OTBR_ERROR_INVALID_ARGS);
+    if (body == "enable")
     {
-        GetDataState(aResponse);
+        if (!otIp6IsEnabled(mInstance))
+            SuccessOrExit(otIp6SetEnabled(mInstance, true));
+        VerifyOrExit(otThreadSetEnabled(mInstance, true) == OT_ERROR_NONE, error = OTBR_ERROR_INVALID_STATE);
+    }
+    else if (body == "disable")
+    {
+        VerifyOrExit(otThreadSetEnabled(mInstance, false) == OT_ERROR_NONE, error = OTBR_ERROR_INVALID_STATE);
     }
     else
     {
+        ExitNow(error = OTBR_ERROR_INVALID_ARGS);
+    }
+
+    errorCode = GetHttpStatus(HttpStatusCode::kStatusOk);
+    aResponse.SetResponsCode(errorCode);
+
+exit:
+    if (error == OTBR_ERROR_INVALID_STATE)
+    {
+        ErrorHandler(aResponse, HttpStatusCode::kStatusConflict);
+    }
+    if (error == OTBR_ERROR_INVALID_ARGS)
+    {
+        ErrorHandler(aResponse, HttpStatusCode::kStatusBadRequest);
+    }
+    else if (error != OTBR_ERROR_NONE)
+    {
+        ErrorHandler(aResponse, HttpStatusCode::kStatusInternalServerError);
+    }
```

Comment:
What about writing a short function to convert otbrError/otError to the response code? For example:
```
int ErrorToResponseCode(otbrError aError) {
    // Converts an otbr error code to a HTTP response code
    // ...
}
```

Then we can have simple error handling code here and everywhere else:
```
exit:
    ErrorHandler(aResponse, ErrorToResponseCode(aError));
```

This will help create consistent response code for every REST APIs

Reply:
There is another downside of this: It makes it harder to reason what codes exactly a particular resource can return (which is quite important, IMHO). That said, it does simplify code quite ab it so :man_shrugging: 

Thoughts? Maybe better in a follow up PR?

Reply:
Fine to leave it as is